### DISCUSSION TITLE
How to have mpv and conky on the same scratchpad?

### DISCUSSION THREAD

**Discussion by ghost** [i3/i3#6230] [Q&A]
I have been trying to get a scratchpad set up with two applications. So far I have it working with one like so:

`bindsym $mod+d scratchpad show`
`exec --no-startup-id conky`
`for_window [class="Conky"] move scratchpad`

This works fine, and I can show and hide conky with `Super+D`

However I also want to be able to open an mpv instance playing music, and have it be part of the same scratchpad.

If I add `--x11-name=floating` to my mpv command, I can then force it onto a scratchpad with:
`for_window [instance="floating"] move scratchpad`

However using `Super+D` doesn't show them both at the same time, it will go `nothing` -> `conky` -> `nothing` -> `mpv` -> `nothing`.

If I change:
`bindsym $mod+d scratchpad show` to
`bindsym $mod+d [class=.*] scratchpad show`

Then they are both on the same scratchpad, and pressing `Super+D` will go `nothing` -> `mpv and conky` -> `nothing`

However there is an issue. If I press `Super+D` to show conky, and then open mpv with my hotkey, it breaks the scratchpads, and pressing `Super+D` will now toggle between mpv and conky, with always one and only one on the screen at the same time.

What is the correct way to do what I'm trying to achieve?

Basically I want a scratchpad which can be toggled with `Super+D`, and the scratchpad should have conky on it (started on launch), and if mpv is open, mpv should also be on the scratchpad, regardless of whether the scratchpad was visible at the time of launching mpv.

**Comment by okraits** [comment]
You didn't mention how you want to have the two applications mpv and conky arranged.

But you can basically have a tiled, tabbed or stacked layout of both mpv and conky, then you need to focus the parent container of those two windows (`$mod+a` in the default config) and then you can send that container to the scratchpad (`$mod+Shift+minus` in the default config).

When you then show the scratchpad, the parent container with both windows will appear.

**Reply by ghost** [reply]
I planned to just have them floating next to each other. mpv in the center and conky at the right of the screen, although the exact position isn't important. 
Basically so it's like an overlay I can show and hide with a hotkey, showing the current music that is playing (mpv), as well as system info (conky)

I think the main problem I am having is running `move to scratchpad` on a window, when you already have a scratchpad displayed, sends it to a different scratchpad. So then `[class=.*] scratchpad show` will just toggle between the two windows, instead of showing and hiding both together.

Basically instead of going from:

```
+----------------------------+
|                            |
|                  +------+  |
|   +------+       |   c  |  |
|   | mpv  |       |   o  |  |
|   |      |       |   n  |  |
|   +------+       |   k  |  |
|                  |   y  |  |
|                  +------+  |
|                            |
+----------------------------+
```

To

```
+----------------------------+
|                            |
|                            |
|                            |
|                            |
|                            |
|                            |
|                            |
|                            |
|                            |
+----------------------------+
```


It will instead go from:

```
+----------------------------+
|                            |
|                  +------+  |
|                  |   c  |  |
|                  |   o  |  |
|                  |   n  |  |
|                  |   k  |  |
|                  |   y  |  |
|                  +------+  |
|                            |
+----------------------------+
```

To:
```
+----------------------------+
|                            |
|                            |
|   +------+                 |
|   | mpv  |                 |
|   |      |                 |
|   +------+                 |
|                            |
|                            |
|                            |
+----------------------------+
```

**Comment by ghost** [comment]
Update:

I decided not to use scratchpads for this, and rather just have a workspace which acts as a desktop. Conky is set to open on this desktop by default:
```
exec --no-startup-id "conky -c ~/.config/conky/conky_i3.conf"
for_window [class="Conky"] move to workspace $ws5, floating enabled
```

In my mpv music script, I added `--x11-name=mainfloating`, and then use this line:
```
for_window [instance="mainfloating"] floating enable, move to workspace $ws5, move position 70 100
```
Which also places those mpv instances on the desktop workspace.

To replicate the "show/hide desktop" functionality I used this script:

```
#!/usr/bin/env bash

current_workspace=$(i3-msg -t get_workspaces | jq -r '.[] | select(.focused==true).name')
#echo "$current_workspace"
past_workspace="1:WWW"
[ -f "$XDG_CACHE_HOME/rc67/i3_last_workspace.txt" ] && past_workspace=$(cat "$XDG_CACHE_HOME/rc67/i3_last_workspace.txt")

if [[ "$current_workspace" == "5:DSK" ]]; then
	i3-msg workspace "$past_workspace"
else
	echo "$current_workspace" > "$XDG_CACHE_HOME/rc67/i3_last_workspace.txt"
	i3-msg workspace "5:DSK"
fi
```

Bound to `Super+d` with:

```
bindsym $mod+d exec --no-startup-id "~/.config/i3/toggle_desktop.sh"
```

The result being a workspace which functions as a desktop that I can switch to and from using `Super+d`.
